org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/internal/gradle/checksums/WrapperValidator.java

```diff
@@ -253,14 +253,24 @@ private static String read(File file) {
 	}
 
 	private static File getVersionCacheFile() {
-		return new File(System.getProperty("user.home"), ".tooling/gradle/versions.json");
+		String xdgCache = getXdgCache();
+		return new File(xdgCache, "tooling/gradle/versions.json");
+	}
+
+	private static String getXdgCache() {
+		String xdgCache = System.getenv("XDG_CACHE_HOME");
+		if (xdgCache == null) {
+			xdgCache = System.getProperty("user.home") + "/.cache/";
+		}
+		return xdgCache;
 	}
 
 	public static File getSha256CacheFile() {
 		String checksumCache = System.getProperty("gradle.checksum.cacheDir");
 		File file;
 		if (checksumCache == null || checksumCache.isEmpty()) {
-			file = new File(System.getProperty("user.home"), ".tooling/gradle/checksums");
+			String xdgCache = getXdgCache();
+			file = new File(xdgCache, "tooling/gradle/versions.json");
```

Comment:
Shouldn't be `checksum` instead of `versions`?

Reply:
You are right. Fixed